libbeat/dashboards/decode.go

```diff
@@ -47,25 +47,31 @@ func DecodeExported(exported []byte) []byte {
 		line, err := r.ReadBytes('\n')
 		if err != nil {
 			if err == io.EOF {
-				return result
+				return append(result, decodeLine(line)...)
 			}
 			return exported
 		}
-		o := common.MapStr{}
-		err = json.Unmarshal(line, &o)
+		result = append(result, decodeLine(line)...)
+	}
+}
+
+func decodeLine(line []byte) []byte {
```

Comment:
I think json.Unmarshal will return an error if you try to decode an empty slice.

You might want to do a `if len(bytes.TrimSpace(line)) == 0 { return }`. This handles both empty slices and blank lines (which are allowable per the NDJSON spec you linked to.)

Reply:
:+1: I have added this check to more functions that needed it.